django/db/backends/sqlite3/base.py

```diff
@@ -170,6 +170,18 @@ def get_connection_params(self):
                 RuntimeWarning,
             )
         kwargs.update({"check_same_thread": False, "uri": True})
+        # Pop `transaction_mode` from kwargs if it exists since it isn't a
+        # valid option for Database.connect().
+        transaction_mode = kwargs.pop("transaction_mode", None)
+        if transaction_mode is not None and transaction_mode.upper() not in (
+            "DEFERRED",
+            "IMMEDIATE",
+            "EXCLUSIVE",
+        ):
+            raise ImproperlyConfigured(
+                "settings.DATABASES is improperly configured. "
+                "OPTIONS['transaction_mode'] value is invalid."
+            )
```

Comment:
You might want to include the value of `transaction_mode` in the error message to help diagnose 

```suggestion
            raise ImproperlyConfigured(
                f"settings.DATABASES[{self.alias!r}]['OPTIONS']['transaction_mode'] is improperly configured. "
                f"{transaction_mode} must be one of {self.allowed_transaction_modes!r}."
            )
```

Reply:
Great idea 👍